### PR TITLE
junkicide/raw bytes from storage

### LINKED ISSUE #836: ledger client: add query for raw bytes from storage
Add a CLI query command to find hex encoded bytes of a given storage key

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201757896515792/1201762707317609) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by junkicide**
closes #836

### CODE REVIEW

File: apps/src/lib/cli.rs [hunk 1]
```diff
@@ -1996,7 +2021,27 @@ pub mod args {
             )
         }
     }
+    /// Query the raw bytes of given storage key
+    #[derive(Clone, Debug)]
+    pub struct QueryRawBytes {
+        /// Common query args
```
Comment (tzemanovic) [inline apps/src/lib/cli.rs:2027]:
```suggestion
        /// The storage key to query
```

File: shared/src/types/storage.rs [hunk 2]
```diff
@@ -185,7 +185,13 @@ impl From<DbKeySeg> for Key {
         }
     }
 }
+impl FromStr for Key {
```
Comment (tzemanovic) [inline shared/src/types/storage.rs:188]: (just a formatting nit, sadly rustfmt doesn't do this)
```suggestion

impl FromStr for Key {
```

File: shared/src/types/storage.rs [hunk 3]
```diff
@@ -185,7 +185,13 @@ impl From<DbKeySeg> for Key {
...
+    type Err = Error;
 
+    fn from_str(s: &str) -> Result<Self> {
+        Key::parse(s)
+    }
+}
```
Comment (tzemanovic) [inline shared/src/types/storage.rs:194]:
```suggestion
}

```

**Review by tzemanovic — APPROVED** [review]
LGTM, thanks!

**Comment by juped** [comment]
rewrote branch for better history

**Comment by juped** [comment]
stuff should really not be merged to master without living on an intermediate branch for a while anyway, though having a chance to fix style is orthogonal to that

**Pull request merged**